src/Plotting/StrengthEnvelope.jl

```diff
@@ -0,0 +1,319 @@
+using Parameters
+using SpecialFunctions: erfc
+using .GLMakie
+
+export StrengthEnvelope, ConstantTemp, LinearTemp, HalfspaceCoolingTemp, GP_Compute_ThermalStructure, LithPres
+
+abstract type AbstractThermalStructure end  
+
+"""
+    ConstantTemp(T=1000)
+    
+Sets a constant temperature inside the box
+Parameters
+===
+- T : the value
+"""
+@with_kw_noshow mutable struct ConstantTemp <: AbstractThermalStructure
+    T = 1000
+end
+
+function GP_Compute_ThermalStructure(Z, s::ConstantTemp)
```

Comment:
This can simply be a one-liner `GP_Compute_ThermalStructure(Z, s::ConstantTemp) = fill(s.T, length(Z)`

Reply:
> GLMakie works on GPUs. As @boriskaus points out, try CairoMakie as an alternative, but double-check first if all the interactive GUI widgets also work (or at least don't crash)

I checked and as Ludo pointed out, CairoMakie cannot create the interactive plot. As the test does not need to be interactive, I thought I could put this at the start of the StrengthEnvelope function:

if mode == "normal"
        using GLMakie
elseif mode == "test"
        using CairoMakie
end

but this also crashes because the Makie functionalities are not loaded yet when it tries to compile the function.

Reply:
> I am wondering whether we should or can even have a test for this. Looking at how Makie does testing it wasn't immediately obvious to me whether they are even doing this for widgets...

I was also wondering if I should just remove the test and hope that it never breaks or instead write a function that is a clone but does not use any of the makie functionality and only default values for testing. But this would be annoying to maintain.